### PR TITLE
Add derive macro for num_traits::Signed

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

**Comment by tdelabro** [comment]
@cuviper Do you think this could be merged?

**Comment by cuviper** [comment]
Would you be willing to write a corresponding newtype-derived `Unsigned` as well? That one is just a marker type, but it could still be useful to have `#[derive(Unsigned)]`, just like you can with the `Copy` marker.

**Comment by tdelabro** [comment]
Sure. I will in the next days

**Comment by tdelabro** [comment]
Hey, @cuviper. Sorry it took so long, I got really busy with my job.

Here it is tho :)

**Comment by tdelabro** [comment]
@cuviper ping

File: tests/newtype.rs [hunk 1]
```diff
@@ -2,9 +2,22 @@ extern crate num as num_renamed;
...
+use crate::num_renamed::{
+    Float, FromPrimitive, Num, NumCast, One, Signed, ToPrimitive, Unsigned, Zero,
+};
 use std::ops::Neg;
 
+#[derive(PartialEq, Zero, One, NumOps, Num, Unsigned)]
+struct MyNum(u32);
+
+#[test]
+fn test_derive_unsingned_works() {
```
Comment (cuviper) [inline tests/newtype.rs]:
```suggestion
fn test_derive_unsigned_works() {
```

File: tests/newtype.rs [hunk 2]
```diff
@@ -87,3 +101,8 @@ fn test_num() {
 fn test_float() {
     assert_eq!(MyFloat(4.0).log(MyFloat(2.0)), MyFloat(2.0));
 }
+
+#[test]
+fn test_signed() {
+    assert_eq!(MyFloat(-2.0).is_negative(), true)
```
Comment (cuviper) [inline tests/newtype.rs]:
```suggestion
    assert!(MyFloat(-2.0).is_negative())
```

**Review by cuviper — COMMENTED** [review]
Just a couple of small things in the tests...

**Pull request merged**